byterun/floats.c

```diff
@@ -66,6 +76,57 @@ CAMLexport void caml_Store_double_val(value val, double dbl)
 
 #endif
 
+/*
+ OCaml runtime itself doesn't call setlocale, i.e. it is using
+ standard "C" locale by default, but it is possible that
+ thirt-party code loaded into process does.
+*/
+#ifdef HAS_LOCALE
+
+#ifdef _MSC_VER
+
+#ifndef locale_t
+#define locale_t _locale_t
+#endif
+
+#ifndef freelocale
+#define freelocale _free_locale
+#endif
+
+#ifndef strtod_l
+#define strtod_l _strtod_l
+#endif
+
+#endif
+
+locale_t caml_locale = 0;
```

Comment:
Isn't `locale_t` a pointer type? I'd much rather make that explicit and use `NULL` instead of `0` (here and in a few other places).